### PR TITLE
Dockerize / php 7

### PR DESCRIPTION
**Pull request by pivnicek**
Switch to php 7 and strict typing and run tests on docker

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,25 +1,48 @@
...
-        "phpunit/phpunit": "3.7.*"
+        "phpunit/phpunit": "7.*",
+        "keboola/coding-standard": "^7.0",
+        "phpstan/phpstan": "^0.8"
     },
     "autoload": {
-        "psr-0": {
-            "Retry": "src"
+        "psr-4": {
+            "Retry\\": "src/Retry/"
```
Comment (odinuv) [inline composer.json]: i'd move the files to src dir instead

File: composer.json [hunk 2]
```diff
@@ -1,25 +1,48 @@
...
         }
+    },
+    "autoload-dev": {
+        "psr-4": {
+            "Retry\\Test\\": "tests/Retry/Test/"
```
Comment (odinuv) [inline composer.json]: also here we can simplify the dir structure

File: composer.json [hunk 3]
```diff
@@ -1,25 +1,48 @@
...
     ],
     "require": {
-        "php": ">=5.4"
+        "php": ">=7.1"
     },
     "require-dev": {
-        "phpunit/phpunit": "3.7.*"
+        "phpunit/phpunit": "7.*",
+        "keboola/coding-standard": "^7.0",
+        "phpstan/phpstan": "^0.8"
```
Comment (odinuv) [inline composer.json]: "phpstan/phpstan-shim": "^0.10"

File: .travis.yml [hunk 4]
```diff
@@ -1,12 +1,8 @@
...
-    - 5.6
-
-before_script:
-    - wget http://getcomposer.org/composer.phar
-    - php composer.phar install --dev
-
-script: phpunit --coverage-text
\ No newline at end of file
+language: bash
+services:
```
Comment (odinuv) [inline .travis.yml]: update badge in readme - it points to vkravity

**Comment by pivnicek** [comment]
Ok, I'm going to merge this then.

**Pull request merged**